src/test/kotlin/study/DslTest.kt

```diff
@@ -0,0 +1,104 @@
+package study
+
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.Test
+
+class DslTest {
+    @Test
+    fun dslTest() {
+        val person = introduce {
+            name("inhyeokk")
+            company("kakao")
+            skills {
+                soft("A passion for problem solving")
+                soft("Good communication skills")
+                hard("Kotlin")
+            }
+            languages {
+                "Koreans" level 5
+                "English" level 3
+            }
+        }
+
+        with(person) {
+            name shouldBe "inhyeokk"
+            company shouldBe "kakao"
+            skills shouldBe listOf(
+                Skill.Soft("A passion for problem solving"),
+                Skill.Soft("Good communication skills"),
+                Skill.Hard("Kotlin")
+            )
+            languages shouldBe listOf(
+                Language("Koreans", 5),
+                Language("English", 3)
+            )
+        }
+    }
+}
+
+fun introduce(block: Person.Builder.() -> Unit): Person = Person.Builder().apply(block).build()
+
+data class Person(val name: String, val company: String, val skills: Skills, val languages: Languages) {
+
+    class Builder {
+        private var name: String = ""
+        private var company: String = ""
```

Comment:
default value가 있는 경우라면 지금처럼 사용하실 수도 있을 것 같아요. 😃
하지만 만약 default value가 없는 필수 값이라면 어떻게 해야할까요? 

필수값들은 `lateinit var`로 지정해서 
```suggestion
        private lateinit var name: String
        private lateinit var company: String
```

<img width="1323" alt="image" src="https://github.com/next-step/kotlin-blackjack/assets/47850258/19fa597a-78ef-4dec-9a66-1a0c7125bb72">

Reply:
기본 값이 없는 경우 남겨주신 것처럼 lateinit var 로 지정할 수도 있긴한데, 캡처로 남겨주신 것처럼 예외가 발생할 수 있어서,

저의 경우 예외가 발생할 수 있는 상황보다는 코드 상에서 필수 값임이 드러날 수 있도록, 필수 값이 지정되어야하는 경우에는 Builder 생성자의 필수 파라미터로 아래와 같이 구현하여 코드 상에서 필수 값으로 전달할 수 있도록 강제하는 편입니다.

- 필수 값의 경우 파라미터로 정의, 선택 값의 경우 Builder 로 정의.

예시 코드 네이밍이 적절하지 않긴한데 대략적인 맥락만 참고해주세요.

```kotlin
fun introduce(name: String, block: Person.Builder.() -> Unit): Person = Person.Builder(name).apply(block).build()

data class Person(val name: String, val company: String, val skills: Skills, val languages: Languages) {

    class Builder(private val name: String) { // 생성자의 필수 파라미터
        private var company: String = ""
        private var skills: Skills = Skills(emptyList())
        private var languages: Languages = Languages(emptyList())

        fun company(company: String) {
            this.company = company
        }

        '''
```

그런데 테스트 가능한 상황을 만들어야 하니까 예외를 발생하는게 더 적절할 것 같기도 하네요.